packages/docusaurus-theme-classic/src/theme/NavbarItem/DefaultNavbarItem.tsx

```diff
@@ -67,21 +67,28 @@ function NavItemDesktop({
   className,
   ...props
 }: DesktopOrMobileNavBarItemProps) {
-  const dropDownRef = React.useRef<HTMLDivElement>(null);
-  const dropDownMenuRef = React.useRef<HTMLUListElement>(null);
-  const [showDropDown, setShowDropDown] = useState(false);
-  useOnClickOutside(dropDownRef, () => toggle(false));
-  function toggle(state: boolean) {
-    if (state) {
-      const firstNavLinkOfULElement =
-        dropDownMenuRef?.current?.firstChild?.firstChild;
-
-      if (firstNavLinkOfULElement) {
-        (firstNavLinkOfULElement as HTMLElement).focus();
+  const dropdownRef = useRef<HTMLDivElement>(null);
+  const dropdownMenuRef = useRef<HTMLUListElement>(null);
+  const [showDropdown, setShowDropdown] = useState(false);
+
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (!dropdownRef.current || dropdownRef.current.contains(event.target)) {
+        return;
       }
-    }
-    setShowDropDown(state);
-  }
+
+      setShowDropdown(false);
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+    document.addEventListener('touchstart', handleClickOutside);
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+      document.removeEventListener('touchstart', handleClickOutside);
+    };
+  }, [dropdownRef]);
```

Comment:
don't like much making the code a bit more complex, but as it should reduce a bit the bundle size, let's do this.